### PR TITLE
add docs for v5.1: Develop -> TiKV Clients

### PR DESCRIPTION
**Pull request by marsishandsome**
### What is changed?

TiKV's official website is planned to be refactored as discussed in https://github.com/tikv/website/issues/227.

This PR covers `Develop` -> `TiKV Clients`

### Any related PRs or issues?

https://github.com/tikv/website/issues/227.

### Which version does your change affect?
v5.1

### CODE REVIEW

**Comment by marsishandsome** [comment]
@andylokandy @zz-jason @iosmanthus PTAL

File: content/docs/5.1/develop/clients/go.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+---
+title: Go Client
+description: Interact with TiKV using Go.
+menu:
+    "5.1":
+        parent: TiKV Clients
+        weight: 2
+---
+
+TiKV client for go is still in the stage of prove-of-concept and under heavy development. You can track development at [tikv/client-go](https://github.com/tikv/client-go/) repository.
```
Comment (iosmanthus) [inline content/docs/5.1/develop/clients/go.md:10]: Seems https://github.com/tikv/client-go/tree/v2 is available.

File: content/docs/5.1/develop/clients/introduction.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+{{< warning >}}
+It is **not recommended or supported** to use both the raw and transactional APIs on the same keyspace.
+{{< /warning >}}
+
+TiKV has clients for a number of languages:
+
+| Clients                    | RawKV API         | TxnKV API         | Supported TiKV Version |
+| -------------------------- | ----------------- | ----------------- | ---------------------- |
+| [Java Client](../java)     | Stable            | Under development | >= 2.0.0               |
+| [Go Client](../go)         | Under development | Under development | >= 5.0.0               |
```
Comment (iosmanthus) [inline content/docs/5.1/develop/clients/introduction.md:26]: using `Unstable` is enough?

File: content/docs/5.1/develop/clients/go.md (hunk 1 above)
Comment (marsishandsome) [reply content/docs/5.1/develop/clients/go.md:10]: menglong suggested to update this document until v2 is released.

File: content/docs/5.1/develop/clients/introduction.md (hunk 2 above)
Comment (marsishandsome) [reply content/docs/5.1/develop/clients/introduction.md:26]: Java TxnKV API is under development.
But Go API is unstable (but available now).
It's more accurate.

File: content/docs/5.1/develop/clients/rust.md [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+```rust
+if all_is_good {
+    txn.commit().await?;
+} else {
+    txn.rollback().await?;
+}
+```
+
+These functions also have batch variants (`batch_put`, `batch_get`, `batch_delete` and `batch_scan`), which offer considerably reduced network overhead and can result in dramatic performance increases under certain workloads.
```
Comment (iosmanthus) [inline content/docs/5.1/develop/clients/rust.md:138]:
```suggestion
These functions also have batch variants (`batch_put`, `batch_get`, `batch_delete` and `batch_scan`), which offer considerably reduced network overhead and can result in dramatic performance improvement under certain workloads.
```

File: content/docs/5.1/develop/clients/rust.md [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+// put
+let mut txn = txn_client.begin_optimistic().await?;
+txn.put(key.to_owned(), value.to_owned()).await?;
+txn.commit().await?;
+
+// get
+let mut txn = txn_client.begin_optimistic().await?;
+let result = txn.get(key.to_owned()).await?;
+txn.commit().await?;
+assert_eq!(result.unwrap(), value.clone().as_bytes());
```
Comment (iosmanthus) [inline content/docs/5.1/develop/clients/rust.md:100]: Could we remove the `clone` here?

File: content/docs/5.1/develop/clients/rust.md (hunk 3 above)
Comment (marsishandsome) [reply content/docs/5.1/develop/clients/rust.md:138]: nice

File: content/docs/5.1/develop/clients/rust.md (hunk 4 above)
Comment (marsishandsome) [reply content/docs/5.1/develop/clients/rust.md]: yes

File: content/docs/5.1/develop/clients/introduction.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+{{< warning >}}
+It is **not recommended or supported** to use both the raw and transactional APIs on the same keyspace.
+{{< /warning >}}
+
+TiKV has clients for a number of languages:
+
+| Clients                    | RawKV API         | TxnKV API         | Supported TiKV Version |
+| -------------------------- | ----------------- | ----------------- | ---------------------- |
+| [Java Client](../java)     | Stable            | Under development | >= 2.0.0               |
+| [Go Client](../go)         | Unstable          | Unstable          | >= 5.0.0               |
```
Comment (zz-jason) [inline content/docs/5.1/develop/clients/introduction.md:26]: should we recommend using client-go v2?

File: content/docs/5.1/develop/clients/java.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    "5.1":
+        parent: TiKV Clients
+        weight: 1
+---
+
+This guide introduces how to interact with TiKV using [Java Client](https://github.com/tikv/client-java).
+
+{{< warning >}}
+TiKV Java Client is developed and released using Java8. The minimum supported version of TiKV is 2.0.0.
+{{< /warning >}}
```
Comment (zz-jason) [inline content/docs/5.1/develop/clients/java.md:14]: why wrapping it with a warning? I think a note is sufficient.

File: content/docs/5.1/develop/clients/java.md [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+
+## Add the dependency
+
+To start, open the `pom.xml` of your project, and add the `tikv-client-java` as dependencies if you are using Maven.
+
+```xml
+<dependency>
+  <groupId>org.tikv</groupId>
+  <artifactId>tikv-client-java</artifactId>
+  <version>${latest_version}</version>
```
Comment (zz-jason) [inline content/docs/5.1/develop/clients/java.md:24]: should we replace ${latest_version} with the later version number?

File: content/docs/5.1/develop/clients/java.md [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+</dependency>
+```
+
+## Raw key-value API
```
Comment (zz-jason) [inline content/docs/5.1/develop/clients/java.md:28]: it's better to illustrate the client-java in these parts:

- a quick introduction and demo show
- the data insertion/deletion operations
- the data retrieving operations
- atomic operations

in this document, we illustrate the functionalities that the java client can provide, helps users to understand what they can do with this client, leave the concert API description to the API Doc.

File: content/docs/5.1/develop/clients/rust.md [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+## Add the dependency
+
+To start, open the `Cargo.toml` of your project, and add the `tikv-client` as dependencies.
+
+```toml
+[dependencies]
+tikv-client = "0.1.0"
+```
+
+## Raw key-value API
```
Comment (zz-jason) [inline content/docs/5.1/develop/clients/rust.md:34]: ditto

File: content/docs/5.1/develop/clients/introduction.md [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+
+TiKV offers two APIs that you can interact with:
+
+| API           | Description                                                                           | Atomicity     | Use when...                                                                      |
+|:------------- |:------------------------------------------------------------------------------------- |:------------- |:-------------------------------------------------------------------------------- |
+| Raw           | A lower-level key-value API for interacting directly with individual key-value pairs. | Single key    | Your application requires low latency and doesn't need distributed transactions. |
+| Transactional | A higher-level key-value API that provides ACID semantics.                            | Multiple keys | Your application requires distributed transactions.                              |
+
+{{< warning >}}
+It is **not recommended or supported** to use both the raw and transactional APIs on the same keyspace.
```
[the remark this answers is not in the document]
Comment (marsishandsome) [reply content/docs/5.1/develop/clients/introduction.md:18]: May affect the data correctness.

File: content/docs/5.1/develop/clients/introduction.md (hunk 5 above)
Comment (marsishandsome) [reply content/docs/5.1/develop/clients/introduction.md:26]: yes, v2 is recommended (after it is released). v1 is only maintained for bug fixes.

File: content/docs/5.1/develop/clients/java.md (hunk 6 above)
Comment (marsishandsome) [reply content/docs/5.1/develop/clients/java.md:14]: sure

File: content/docs/5.1/develop/clients/java.md (hunk 7 above)
Comment (marsishandsome) [reply content/docs/5.1/develop/clients/java.md:24]: If we use the latest version number here, it must be updated each time a new version is released.

Comment (marsishandsome) [reply content/docs/5.1/develop/clients/java.md:24]: Another problem, the example code here uses the API in version `3.1.0-SNAPSHOT` which is not released yet.

Comment (marsishandsome) [reply content/docs/5.1/develop/clients/java.md]: update to `3.2.0-SNAPSHOT`

**Review by zz-jason — APPROVED** [review]

File: content/docs/5.1/develop/clients/java.md (hunk 8 above)
Comment (marsishandsome) [reply content/docs/5.1/develop/clients/java.md:28]: I will put the examples in `Develop -> RawKV` and `Develop -> TxnKV` chapter.

File: content/docs/5.1/develop/clients/cpp.md [hunk 11]
```diff
@@ -0,0 +1,16 @@
+---
+title: C++ Client
+description: Interact with TiKV using C.
```
Comment (iosmanthus) [inline content/docs/5.1/develop/clients/cpp.md:3]:
```suggestion
description: Interact with TiKV using C++.
```

File: content/docs/5.1/develop/clients/cpp.md [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+        parent: TiKV Clients
+        weight: 5
+---
+
+TiKV client for C++ is built on top of [TiKV Client in Rust](https://github.com/tikv/client-rust) via [cxx](https://github.com/dtolnay/cxx).
+
+This client is still in the stage of prove-of-concept and under heavy development. You can track development at [tikv/client-cpp](https://github.com/tikv/client-cpp/) repository.
+
+{{< warning >}}
+You should not use the C++ client for production use until it is released.
```
Comment (iosmanthus) [inline content/docs/5.1/develop/clients/cpp.md:15]:
```suggestion
You are not suggested to use the C++ client for production use until it is released.
```
might be better? You can ignore this comment if you think the old one is better.

File: content/docs/5.1/develop/clients/go.md [hunk 13]
```diff
@@ -0,0 +1,14 @@
...
+
+{{< warning >}}
+You should not use the go client for production use until it is released.
```
Comment (iosmanthus) [inline content/docs/5.1/develop/clients/go.md:13]: Ditto.

**Comment by iosmanthus** [comment]
Rest LGTM.

**Pull request merged**